client/my-sites/stats/stats-notices/index.tsx

```diff
@@ -22,6 +39,28 @@ const NewStatsNotices = ( { siteId, isOdysseyStats }: StatsNoticesProps ) => {
 		isJetpackSite( state, siteId, { treatAtomicAsJetpackSite: false } )
 	);
 
+	const reduxDispatch = useDispatch();
+	// TODO: Display error messages on the notice.
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	const [ error, setError ] = useState< Error | null >( null );
+
+	// Update site purchases in Redux store for Odyssey Stats.
+	useEffect( () => {
+		if ( isOdysseyStats ) {
+			wpcom.req
+				.get( { path: '/site/purchases', apiNamespace: 'jetpack/v4' } )
+				.then( ( res: { data: string } ) => JSON.parse( res.data ) )
+				.then( ( purchases: Record< string, string >[] ) => {
+					reduxDispatch( {
+						type: SITE_PURCHASES_UPDATE,
+						siteId,
+						purchases: formatPurchasesToSiteProducts( purchases ),
+					} );
+				} )
+				.catch( ( error: Error ) => setError( error ) );
+		}
```

Comment:
Would it make sense to refactor this into a hook or a query component in the future?

Reply:
Agreed. I guess it'd be better refactor it into a hook. Another thing is it'll easier to use `react-query` 😄

Reply:
Refactored in [105895f](https://github.com/Automattic/wp-calypso/pull/79314/commits/105895f689f0c8655de2e33e2770cdf215be8b4f).